src/main/java/org/jabref/logic/shared/DBMSSynchronizer.java

```diff
@@ -207,13 +210,16 @@ public void synchronizeLocalDatabase() {
                     }
                 }
             }
-            if (!match) {
-                entriesToDrag.add(idVersionEntry.getKey());
+            if (!remoteEntryMatchingOneLocalEntryFound) {
+                entriesToInsertIntoLocalDatabase.add(idVersionEntry.getKey());
             }
         }
 
-        for (BibEntry bibEntry : dbmsProcessor.getSharedEntries(entriesToDrag)) {
-            bibDatabase.insertEntry(bibEntry, EntriesEventSource.SHARED);
+        if (!entriesToInsertIntoLocalDatabase.isEmpty()) {
+            // in case entries should be added into the local database, insert them
+            for (BibEntry bibEntry : dbmsProcessor.getSharedEntries(entriesToInsertIntoLocalDatabase)) {
+                bibDatabase.insertEntry(bibEntry, EntriesEventSource.SHARED);
```

Comment:
please don't use `insertEntry` in the for loop, but use `insertEntries` for performance reasons.